upload.py

```diff
@@ -73,6 +73,41 @@ def write_to_report(props, text):
     props.report = props.report + " - " + text + "\n\n"
 
 
+def prevalidate_model(props):
+    """ "Check model for possible problems"""
```

Comment:
Extra `"` and missing `.`. Would be nice to describe the actual possible problems this function covers.